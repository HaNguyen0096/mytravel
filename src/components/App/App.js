import React, { Component } from 'react'
import { Route, Switch } from 'react-router-dom'
import Header from '../Header/Header'
import LandingPage from '../../routes/LandingPage/LandingPage'
import PublicPage_MapView from '../../routes/PublicPage_MapView/PublicPage_MapView'
import PublicPage_ListView from '../../routes/PublicPage_ListView/PublicPage_ListView'
import PersonalPage_MapView from '../../routes/PersonalPage_MapView/PersonalPage_MapView'
import PersonalPage_ListView from '../../routes/PersonalPage_ListView/PersonalPage_LisView'
import LogPage from '../../routes/LogPage/LogPage'
import AddLogPage from '../../routes/AddLogPage/AddLogPage'
import LoginPage from '../../routes/LoginPage/LoginPage'
import RegistrationPage from '../../routes/RegistrationPage/RegistrationPage'
import NotFoundPage from '../../routes/NotFoundPage/NotFoundPage'
import PrivateRoute from '../Utils/PrivateRoute'
import PublicRoute from '../Utils/PublicRoute'

//import Footer from '../Footer/Footer';
//import './App.css'


class App extends Component {

  state = { 
    hasError: false 
  }

  static getDerivedStateFromError(error) {
    console.error(error)
    return { hasError: true }
  }

  render() {
    return(
      < >
        <div className='App'>
          <header className='App__header'>
            <Header />
          </header>
        <main className='App__main'>
        {this.state.hasError && <p className='red'>There was an error! Oh no!</p>}
          {
            <Switch>
            <Route 
              exact
              path={'/'}
              component={LandingPage}
            />
            <PublicRoute 
              exact
              path={'/login'}
              component={LoginPage}
            />
            <PublicRoute 
              exact
              path={'/register'}
              component={RegistrationPage}
            />
            <Route
              exact
              path={'/public/mapview'}
              component={PublicPage_MapView}
            />
            <Route
              exact
              path={'/public/listview'}
              component={PublicPage_ListView}
            />
            <PrivateRoute
              exact
              path={'/mylogs/mapview'}
              component={PersonalPage_MapView}
            /> 
            <PrivateRoute
              exact
              path={'/mylogs/listview'}
              component={PersonalPage_ListView}
            />
            <PrivateRoute
              exact
              path={'/logs/:id'}
              component={LogPage}
            /> 
            <PrivateRoute
              exact
              path={'/addlog'}
              component={AddLogPage}
            />
            <Route
              component={NotFoundPage}
            />
            </Switch>
          }
        </main>        
      </div>
      </>     
    )
  }
}

export default App;
