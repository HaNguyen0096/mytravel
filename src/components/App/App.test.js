import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import App from './App'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('App component', () => {

  it('renders App without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <App />
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI for App as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
                <App />
              </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders App by default', () => {
    const wrapper = shallow(<App />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})