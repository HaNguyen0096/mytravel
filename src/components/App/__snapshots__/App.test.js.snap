// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App by default 1`] = `
<Fragment>
  <div
    className="App"
  >
    <header
      className="App__header"
    >
      <Header />
    </header>
    <main
      className="App__main"
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/public/mapview"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/public/listview"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/mylogs/mapview"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/mylogs/listview"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/logs/:id"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/addlog"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </main>
  </div>
</Fragment>
`;

exports[`App component renders the UI for App as expected 1`] = `
<div
  className="App"
>
  <header
    className="App__header"
  >
    <div
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="loginNav"
        >
          <div
            className="title"
          >
            My Travels
          </div>
          <div>
            <a
              className="home-nav"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
            <a
              className="public-nav"
              href="/public/mapview"
              onClick={[Function]}
            >
              Public Logs
            </a>
            <a
              className="my-nav"
              href="/mylogs/mapview"
              onClick={[Function]}
            >
              My Logs
            </a>
          </div>
        </div>
      </div>
      <nav
        className="navBar"
      />
    </div>
  </header>
  <main
    className="App__main"
  >
    <div
      className="landingImage"
    >
      <section
        className="Section"
      >
        <div
          className="landingPage"
        >
          <div
            className="welcomeText"
          >
            Welcome to My Travels
          </div>
          <hr
            className="hrLine"
          />
          <p
            className="landingPage-Script"
          >
            Let's create your own Travels Logs and explore amazing places around the world!
          </p>
        </div>
      </section>
    </div>
  </main>
</div>
`;
